### ISSUE TITLE
Avoid UPPERCASE labels in Wagtail's admin interface

### ISSUE THREAD

**Issue by lb-** [wagtail/wagtail#7624]
### Is your proposal related to a problem?

* Some screen readers will read UPPERCASE as an abbreviation (e.g. space out the word, reading a single letter at a time).
* UI and accessibility guidelines recommend either 'Title Case' or 'Sentence case' for button labels

### Describe the solution you'd like

* Remove all usage of CSS transform to upper case on buttons throughout Wagtail
* Recommendation: Sentence case not UPPERCASE e.g. Publish not PUBLISH

### Describe alternatives you've considered

* Add a screen reader label that has the non-uppercase variant to all buttons, however this could trigger double reading in some cases which might be worse.
* Leave as is

#### Reasons for
 - Avoids issues where screen readers can sometimes read as P.U.B.L.I.S.H (as an abbreviation)
 - Better visual difference across letters
 - Less yelling
 - Capitalisation makes assumptions about language that do not necessarily hold true in other languages (e.g. some languages do not have capitalisation mean the same thing or at all in their language).
 - Makes translation harder, similar to above, but can cause issues where all caps translations ‘leak’ into translations as the UI shows in UPPERCASE when viewed (even though this is done by CSS).
 - Avoids complexity in interpretation, where all caps may end up being in other content that is not a button or visa-versa (e.g. links, table headers that are sortable vs not)
 - Often less horizontal space is taken up by buttons this way meaning a slightly cleaner UI

#### Reasons against

- Material UI (Google) advocates for UPPERCASE BUTTON LABELS
- There is not really 100% consensus on this, nor is there a WCAG guideline for this
- Change - keep with what we are doing

#### Links

 - https://humanmade.github.io/hm-pattern-library/pages/forms.html
 - https://accessibility.huit.harvard.edu/design-readability
 - https://a11y-101.com/design/capitalisation
 - https://uxmovement.com/content/all-caps-hard-for-users-to-read/
 - https://webaim.org/articles/evaluatingcognitive/ > Use stylistic differences to highlight important content, but do so conservatively
 - https://developer.apple.com/design/human-interface-guidelines/ios/controls/buttons/

#### Screenshots

Curtsey of @thibaudcolas 

**Current**

![text_uppercase_before](https://user-images.githubusercontent.com/1396140/137609231-6d26c982-99ac-4f9a-813c-45ade6dbcc17.png)

**Future**

![screenshot_of_the_page_editor_in_wagtail_admin_with_all_uppercase_text_transformations_disabled](https://user-images.githubusercontent.com/1396140/137609235-c4a20057-ac72-47c4-b4b4-06a77f2e9fdb.png)

**Comment by thibaudcolas** [comment]
FYI we will be doing this as part of the [page editor 2022](https://github.com/wagtail/wagtail/discussions/7739) project.

**Comment by thibaudcolas** [comment]
Changing gears – after reviewing the code, I’ll tag this as a `good first issue` that anyone can pick up. This is largely a case of removing all instances of `text-transform: uppercase;` in Wagtail’s stylesheets, so most of the effort will be to set up Wagtail for development and test the different cases, there isn’t a requirement to have deep knowledge of the project.

- There are 22 matches for `text-transform: uppercase` in the Wagtail codebase which can be removed, and 2 which should stay (see below).
- In addition to `text-transform: uppercase`, there are the `label-uppercase` and `u-text-transform-uppercase` CSS classes, which can be removed altogether from the stylesheets and templates.

There are two instances of `text-transform: uppercase` which *shouldn’t* be removed:

- `PublicationStatus.scss` / `c-status` – we want this to remain
- `_status_tag.scss` / `status-tag` – same, we want to keep this

Once we’ve removed what needs to be removed, we will need to add back one `text-transform: uppercase;` to our `.button.button--live` styles in `page-editor.scss`.

**Comment by lb-** [comment]
@thibaudcolas  should we add a linting rule to block additional usage of `text-transform` (maybe on wagtail sctylelint config)

**Comment by thibaudcolas** [comment]
@lb- yes, good idea

**Comment by PaarthAgarwal** [comment]
> Changing gears – after reviewing the code, I’ll tag this as a `good first issue` that anyone can pick up. This is largely a case of removing all instances of `text-transform: uppercase;` in Wagtail’s stylesheets, so most of the effort will be to set up Wagtail for development and test the different cases, there isn’t a requirement to have deep knowledge of the project.
> 
> * There are 22 matches for `text-transform: uppercase` in the Wagtail codebase which can be removed, and 2 which should stay (see below).
> * In addition to `text-transform: uppercase`, there are the `label-uppercase` and `u-text-transform-uppercase` CSS classes, which can be removed altogether from the stylesheets and templates.
> 
> There are two instances of `text-transform: uppercase` which _shouldn’t_ be removed:
> 
> * `PublicationStatus.scss` / `c-status` – we want this to remain
> * `_status_tag.scss` / `status-tag` – same, we want to keep this
> 
> Once we’ve removed what needs to be removed, we will need to add back one `text-transform: uppercase;` to our `.button.button--live` styles in `page-editor.scss`.

I removed all instances of text-transform: uppercase (except 2) as you mentioned but I can't see the reflection of the same in bakerydemo admin page. How to reflect these changes in it?

**Comment by thibaudcolas** [comment]
Hi @PaarthAgarwal, thank you for giving this a go! Do you have our Node tooling up and running? This would be the first thing I would check – since we write styles in Sass, there is a build step to convert this to CSS. See the following setup instructions: https://docs.wagtail.org/en/stable/contributing/developing.html#setting-up-the-wagtail-codebase.

**Comment by PaarthAgarwal** [comment]
Hey @thibaudcolas, I did these steps to setup and build codebase. Then I did these these steps using the same virtual environment I used for building codebase:

**Add Bakery demo (the demo project)**
cd ..
git clone https://github.com/wagtail/bakerydemo.git
cd bakerydemo
cp bakerydemo/settings/local.py.example bakerydemo/settings/local.py
echo "DJANGO_SETTINGS_MODULE=bakerydemo.settings.local" > .env

**Remove the wagtail from bakery requirements.txt. We just installed Wagtail remember?**
sed -i 's/wagtail>=/# wagtail>=/g' requirements/base.txt

**Install bakery requirements**
python -m pip install -r requirements.txt

**Default Django db stuff.**
python manage.py makemigrations
python manage.py migrate

**Sample data.**
python manage.py load_initial_data

python manage.py runserver

**Comment by thibaudcolas** [comment]
@PaarthAgarwal could you point me to where you got those instructions from? The instructions on https://docs.wagtail.org/en/stable/contributing/developing.html#development look quite different, I’d suggest redoing the setup following them (or using the ready-made Docker or Vagrant development setups).

**Comment by PaarthAgarwal** [comment]
@thibaudcolas I setup the codebase using these instructions https://docs.wagtail.org/en/stable/contributing/developing.html#development
Then for setting up bakerydemo I used those instruction mentioned here on slack channel https://wagtailcms.slack.com/archives/CTL4SFX29/p1646832761746769

**Comment by PaarthAgarwal** [comment]
@thibaudcolas I got the fix.
After making changes to scss files I did npm start. Now the admin page will look like this:

![image](https://user-images.githubusercontent.com/86092410/157914462-072474f8-e447-49cb-9030-e383d93d52ea.png)

**Comment by thibaudcolas** [comment]
Nice! 🙌  Those instructions are indeed missing what’s needed to do style and JS changes. Looking forward to your PR!

**Comment by PaarthAgarwal** [comment]
@thibaudcolas my first PR here #8111! I hope it fixes this issue.

**Comment by thibaudcolas** [comment]
Thank you @PaarthAgarwal for taking this on! It’s a pretty solid first contribution 😄  I’ve merged #8111, and https://github.com/wagtail/stylelint-config-wagtail/pull/9, and made a new release of our Stylelint configuration.

I have created #8121 as a follow-up task – please let us know if you want to go for it, if not we’ll see if someone else might pick it up.
